optimum/amd/brevitas/quantizer.py

```diff
@@ -0,0 +1,327 @@
+            else:
+                raise ValueError(
+                    f"Quantization with an FX graph is currently only supported for models taking `input_ids`, `attention_mask` and `past_key_values` as inputs. The model only has the following inputs: {forward_signature}"
+                )
+
+            with torch.no_grad():
+                model = symbolic_trace(self.model, input_names)
+        else:
+            model = self.model
+
+        # Because accelerate is not compatible with FX, we keep two versions of the Model
+        # one with FX-traced, the other one not.
+        # Since weights are shared across the two, we can apply weight/activation equalization
+        # by using one representation or the other based on needs.
+
+        if quantization_config.apply_weight_equalization:
+            logger.info("Applying weight equalization...")
+            apply_weight_equalization(model)
+            logger.info("Weight equalization applied.")
+
+        if quantization_config.activations_equalization is not None:
+            logger.info(
+                f"Applying Activation Equalization {quantization_config.activations_equalization} (SmoothQuant)..."
+            )
+            apply_act_equalization(model, quantization_config.activations_equalization, calibration_dataset)
+            logger.info("Activation equalization applied.")
+
+        # We do not quantize embedding and last fully connected layer
+        model = quantize_model(
+            model,
+            dtype=dtype,
+            weight_quant_format="int",
+            weight_quant_type="sym" if quantization_config.weights_symmetric else "asym",
+            weight_bit_width=quantization_config.weights_bitwidth,
+            weight_param_method=quantization_config.weights_param_method,
+            weight_scale_precision=quantization_config.scale_precision,
+            weight_quant_granularity=quantization_config.weights_quant_granularity,
+            weight_group_size=quantization_config.weights_group_size,
+            quantize_weight_zero_point=quantization_config.quantize_zero_point,
+            input_bit_width=quantization_config.activations_bitwidth,
+            input_quant_type="sym" if quantization_config.activations_symmetric else "asym",
+            input_quant_format="int",
+            input_param_method=quantization_config.activations_param_method,
+            input_scale_precision=quantization_config.scale_precision,
+            input_scale_type="static" if quantization_config.is_static else "dynamic",
+            input_quant_granularity=quantization_config.activations_quant_granularity,
+            input_group_size=quantization_config.activations_group_size,
+            quantize_input_zero_point=quantization_config.quantize_zero_point,
+        )
+
+        # Perform a single inference pass to generate the correct state_dict
+        if quantization_config.apply_gptq or quantization_config.is_static:
```

Comment:
This is necessary every time we need to call offload even in the case of just evaluating the model, so either we use the `calibration_dataset` or a dummy input.

We do not need a real input, just trigger a forward pass.